### PR TITLE
Add further detail to the Implementation chapter

### PR DESCRIPTION
**Pull request by blschatz**
Fleshed out details of this chapter.

### CODE REVIEW

File: implementation/README.md [hunk 1]
```diff
@@ -12,32 +16,35 @@ DXE process is very complex; some systems have 750 DXEs.
...
        *  In production, it has solved problems
+7.	Clear ME region for initrd storage       
+8.  Replace some DXEs with open source components (change by replacement)
 
-1.  Replace some DXEs with open source components (change by replacement)
+One of the challenges in the above is in finding (or reclaiming) enough space in flash to shoehorn your kernel and initrd into. 
 
 ## Tools of the trade
 
 There are two tools you use when you modify the UEFI flash image: `utk `and `me_cleaner`
```
Comment (rminnich) [inline implementation/README.md:45]: if you are going to put me_cleaner first in the text might as well put it first in this sentence.

File: implementation/README.md [hunk 2]
```diff
@@ -12,32 +16,35 @@ DXE process is very complex; some systems have 750 DXEs.
...
 
-The `utk` tool can:
-
-+   Remove DXEs
-+   Insert new DXEs
-+   Replace the binary code of a DXE with a kernel
-
-The Management Engine (ME) is an x86 CPU embedded in the Intel Platform Controller Hub (PCH). It runs the Minix operating system which boots first and enables hardware such as clocks and GPIOs. ME checks the contents of flash memory and is used to implement "BootGuard". If you reflash and the ME is in "BootGuard" mode, your machine will be unusable. You need to run a tool called `me_cleaner` on the image to disable BootGuard.
+The ME Cleaner tool:
```
Comment (rminnich) [inline implementation/README.md:47]: I don't understand why you removed this text.

File: implementation/README.md [hunk 3]
```diff
@@ -95,7 +112,29 @@ Data is provided by TFTP. HTTP downloading takes a fraction of a second even for
 
 Note: Boot images require a kernel(bzImage) + an initramfs + a command line. They can be loaded as three pieces or compiled and loaded as one piece, as described in this section.
 
-### Step 2: replace Shell binary section
+### Step 2: read & write the flash
+
```
Comment (rminnich) [inline implementation/README.md:116]: This is very nice!

File: implementation/README.md (hunk 1 above)
Comment (blschatz) [reply implementation/README.md:45]: Which sentence?

**Review by rjoleary — APPROVED** [review]

**Pull request merged**